wfe2/wfe.go

```diff
@@ -555,6 +559,17 @@ func (wfe *WebFrontEndImpl) NewAccount(
 		InitialIP: ip,
 	})
 	if err != nil {
+		if berrors.Is(err, berrors.Duplicate) {
+			existingAcct, err := wfe.SA.GetRegistrationByKey(ctx, key)
```

Comment:
Similarly, I'd like to reduce complexity here by deleting the preexisting GetRegistrationByKey call and just relying on the new `Duplicate` behavior.

Reply:
We can't remove the existing check as it'll break `OnlyReturnExisting` if we only rely on an error from the insert.

Reply:
Oh good call, thanks. :+1: